plugins/linux-pipewire/pipewire.c

```diff
@@ -874,25 +920,77 @@ uint32_t obs_pipewire_get_width(obs_pipewire_data *obs_pw)
 	if (!obs_pw->negotiated)
 		return 0;
 
-	if (obs_pw->crop.valid)
-		return obs_pw->crop.width;
-	else
-		return obs_pw->format.info.raw.size.width;
+	if (obs_pw->crop.valid) {
```

Comment:
Maybe store the possible width and height based on crop in a variable and then choose the correct one from a single switch to avoid duplicating this switch twice and reduce nesting.

Reply:
I changed it to a ternary operator and it doesn't look as bad as I thought. If you think otherwise I can go with intermediate variables.